ipa-core/src/helpers/mod.rs

```diff
@@ -245,6 +245,17 @@ pub enum Direction {
     Right,
 }
 
+/// implements negation for `Direction`
+impl std::ops::Neg for Direction {
```

Comment:
Oh that may be too magical, maybe just add `fn other(&self) -> Self` to the `Direction` impl?